pkg/encrypt/server-side.go

```diff
@@ -0,0 +1,146 @@
+/*
+ * Minio Go Library for Amazon S3 Compatible Cloud Storage
+ * Copyright 2018 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package encrypt
+
+import (
+	"crypto/md5"
+	"encoding/base64"
+	"errors"
+	"net/http"
+
+	"golang.org/x/crypto/argon2"
+)
+
+const (
+	// sse is the AWS SSE header used for SSE-S3 and SSE-KMS.
+	sse = "X-Amz-Server-Side-Encryption"
```

Comment:
I think it could be better if we rename these constats to little more meaningful like sseHdr, sseHdrCustomerAlgorithm, etc..

What do you think @aead ?

Reply:
The constants are taken from the server code (but unreported). `sse` is the only one I added directly - but I agree that we can use a more expressive name for `sse`. Maybe `sseMethod` / `sseType`.